src/shell.zig

```diff
@@ -201,16 +211,18 @@ pub fn exec_stdout(shell: *Shell, comptime cmd: []const u8, cmd_args: anytype) !
 }
 
 /// Returns current git commit hash as an ASCII string.
-pub fn git_commit(shell: *Shell) ![40]u8 {
+pub fn git_commit(shell: *Shell) ![]const u8 {
+    try shell.exec("git fetch --tags", .{});
```

Comment:
fetching tags from the network is a slow, failable operation. Given that we run this on every build, it seems better to avoid maybe ask the caller to fetch the repo with tags instead?

Reply:
> Given that we run this on every build

Yes, however, locally it's going to rarely do anything. Do you still think we should drop it?

```
$ time git fetch --tags
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 16 (delta 10), reused 13 (delta 10), pack-reused 0
Unpacking objects: 100% (16/16), 5.68 KiB | 528.00 KiB/s, done.
From github.com:tigerbeetledb/tigerbeetle
 + 77184c64...b87c81c8 batiati-eytzinger                                                       -> origin/batiati-eytzinger  (forced update)
 * [new branch]        gh-readonly-queue/main/pr-1140-6378bbe359c0ae2721344ac867ac521f2e5db66a -> origin/gh-readonly-queue/main/pr-1140-6378bbe359c0ae2721344ac867ac521f2e5db66a
 + 80a01ed5...16fd8150 matklad/alligned-free                                                   -> origin/matklad/alligned-free  (forced update)
 + c5ec87c0...8f9bbff5 matklad/tmp-tigerbeetle                                                 -> origin/matklad/tmp-tigerbeetle  (forced update)
git fetch --tags  0.08s user 0.05s system 18% cpu 0.688 total
$ time git fetch --tags
git fetch --tags  0.07s user 0.04s system 18% cpu 0.620 total
$ time git fetch --tags
git fetch --tags  0.07s user 0.04s system 18% cpu 0.593 total
$ time git fetch --tags
git fetch --tags  0.07s user 0.04s system 18% cpu 0.572 total
```

Reply:
Ok! I'll move it into the ci steps then.

Reply:
I guess there's also a considiration that we want `zig build` to work if there's no internet connection.

Reply:
That will depend on how you cloned it then. Since although we clone in GitHub actions, it doesn't have any tags and got describe --tags fails, so the build fails.

However, if we drop the hard check then it can be easy to reintroduce this bug of TigerBeetle version not being set correctly. Unless we add new tests for after a release.

Reply:
FWIW, that's what I use to fetch tags to gh builders:

https://github.com/matklad/once_cell/blob/3cd65494663f003b9b2e0be5b37b3426e543f72a/.github/workflows/ci.yaml#L22-L23

Maybe there's a better way, I don't remeber how I came up with that snippet

Reply:
Ok!

Reply:
Well now it started succeeding without me needing to fetch anything. Wonderful! That means it may fail sometime in the future. But I'm not sure which steps would fail.

Reply:
Nevermind! It's `git tag --describe --always` that handles it without needing to fetch. We're all set.